common-utils/src/common/deployment/__init__.py

```diff
@@ -356,125 +378,270 @@ class FlowSpec(BaseModel):
     """This is the object we use to deploy a flow."""
 
     flow: Flow = Field(..., description="Flow funciton object.")
-    secrets: list[FlowSecret] = Field(
-        [], description="List of FlowSecret objects to pass to task definition."
+    secrets: list[FlowEnvar] = Field(
+        [],
+        description=(
+            "List of secrets as FlowEnvar objects to pass to task definition."
+            "These should only be the secret names and not full arn."
+        ),
     )
     docker_env: StrictStr = Field(
         ...,
-        description="docker environment name from pyproject.toml, which is the key name in '[tool.prefect.envs.base]'",
+        description="docker environment name from pyproject.toml, which is the key name in '[tool.prefect.envs.<>]'",
     )
     ephemeral_storage_gb: conint(gt=19, lt=201) = 20  # type: ignore
     deployments: list[FlowDeployment] = []
-    _slugified_flow_name: StrictStr = PrivateAttr()
-    _slugified_fn_name: StrictStr = PrivateAttr()
-    _project_name: StrictStr = PrivateAttr()
 
     class Config:
         arbitrary_types_allowed = True
 
     def __init__(self, **data) -> None:
-        """Set the private fields on intansiation.
-        DISABLE_FLOW_SPEC will be set to true to disable validation for ECS execution.
+        """Set the flow name on deployment."""
+        super().__init__(**data)
+        if x := os.getenv("POCKET_PREFECT_FLOW_NAME"):
+            self.flow.name = x
+
+    def _handle_task_definition(
+        self, account_id: str, ecs_client: object, slugified_flow_name: str
+    ) -> str | None:
+        """Register Task Defintition with AWS ECS as needed based on changes.
+
+        Args:
+            account_id (str): AWS account id
+            ecs_client (object): valid ecs client
+            slugified_flow_name (str): slugified flow name for naming
+            of task definition and block
+
+        Returns:
+            str: New or existing Task Definition ARN.
         """
-        if DISABLE_FLOW_SPEC != "true":
-            super().__init__(**data)
-            project_name = PYPROJECT_METADATA.project_name
-            slugified_fn_name = standard_slugify(self.flow.fn.__name__)
-            self._project_name = project_name
-            self._slugified_fn_name = slugified_fn_name
-            self.flow.name = f"{project_name}.{slugified_fn_name}"
-
-    @validator("docker_env")
-    def docker_env_must_be_registered(cls, v):
-        docker_envs = PYPROJECT_METADATA.docker_envs
-        if v not in docker_envs.keys():
-            raise ValueError(
-                "Docker env does not exist in pyproject.toml.  It must be the key name in a '[tool.prefect.envs.<key name>]' configuration."
```

Comment:
this is all deprecated as part of reducing external dependencies from the FLOW_SPEC instantiation